### DISCUSSION TITLE
Pulsar Plus Cable Connected always unplugged?

### DISCUSSION THREAD

**Discussion by petervk** [jagheterfredrik/wallbox-mqtt-bridge#62] [General]
Hi, I have a rooted (thank you!) pulsar plus running mqtt-bridge as well as wallbox firmware 6.6.5 and the cable connected sensor is always unplugged. Is there any way to get this to work?

**Comment by lostOzone** [comment]
Check this post
https://community.home-assistant.io/t/wallbox-pulsar-plus-integration/200339/1195

**Reply by petervk** [reply]
Is there any work on getting the mqtt-bridge working with the latest firmware?

**Comment by petervk** [comment]
So I ssh'd into my wallbox and used redis-cli to enter interactive mode and then ran "MONITOR" for a few seconds to see what was produced, and I see "SET" commands like this: 
`1726252203.865875 [0 127.0.0.1:38660] "SET" "wallbox:wallboxsmachine::state" "B6" `
But if I review wallbox_const.go it has all the stateMachinesStates starting with 0x. 
I'm assuming the cable connected status is already taken from redis based on the code I see in wallbox.go.
So is it just that the mqtt-bridge is expecting the format of the data in redis to be different than what wallbox firmware 6.6 does?

**Reply by jagheterfredrik** [reply]
I don’t have 6.6. Can you post more from `MONITOR`?

**Reply by petervk** [reply]
[2024-09-14 - wallbox redis export.txt](https://github.com/user-attachments/files/17003265/2024-09-14.-.wallbox.redis.export.txt)

**Reply by petervk** [reply]
I didn't redact anything but let me know if there is anything private in there that I didn't recognize/catch.

**Reply by jagheterfredrik** [reply]
Thanks, it looks like a lot of work unfortunately. It seems Wallbox changed things up quite a bit, like @tronikos mentioned in the home assistant forums linked further up. I think the biggest issue I see is that it seems very difficult to make an implementation that works for both <6.6 and >=6.6 😢

**Reply by petervk** [reply]
Anything I can do to help?

**Reply by jagheterfredrik** [reply]
Does your wallbox restore to 6.6?

**Reply by petervk** [reply]
No, it came with v5.something

**Reply by jagheterfredrik** [reply]
Then simply restore and use it with the mqtt bridge?

**Reply by sincze** [reply]
> Then simply restore and use it with the mqtt bridge?

Is an option I guess, but if somehow we can make it future proof.  :) 

New boxes can not be rooted anymore??

My wallbox app thought about an auto update ....  to my knowledge, I did not switch that option on.

**Reply by petervk** [reply]
> Then simply restore and use it with the mqtt bridge?

Well yes I could, but wouldn't it be better if the mqtt-bridge supported the latest firmware?

**Reply by sincze** [reply]
It would be even better if wallbox would support the mqtt-bridge by default. 

It does not harm the box. 
It can boost sales for privacy or automation minded buyers.
....
